e2e/cypress/integration/plugins/marketplace/marketplace_spec.js

```diff
@@ -22,14 +22,16 @@ describe('Plugin Marketplace', () => {
 
     describe('should not render in main menu', () => {
         afterEach(() => {
-            // # Click hamburger main menu
-            cy.get('#sidebarHeaderDropdownButton').click();
-
-            // * Dropdown menu should be visible
-            cy.get('#sidebarDropdownMenu').should('be.visible');
-
-            // * Marketplace button should not be visible
-            cy.get('#marketplaceModal').should('not.be.visible');
+            cy.get('#lhsHeader').should('be.visible').within(() => {
```

Comment:
The Cypress [documentation for within](https://docs.cypress.io/api/commands/within.html#Syntax) says:
> Scopes all subsequent cy commands to within this element. Useful when working within a particular group of elements such as a <form>.

This doesn't /seem/ to be how `within` was meant to be used -- in particular, we don't seem to ever use the scoped element passed to the callback. Why isn't just just flat, with the newly added `#lhsHeader` check that presumably fixes the issue?

Reply:
Scope is based on whether it belongs to the same parent element. Per my observation, there are a number of advantages in doing scope/within, especially when dealing with flaky tests:
- Narrow down selection. Scan of child element (target) is scoped into a particular structure of the app and not the entire document
- Allow target element to fully render in time. Only scan child element when parent element is visible (check for visibility has wait under the hood and in effect, it "may buy" some time to render children before doing actual assertion)
- Specifically target the element. It eliminates getting multiple elements (in case a selector is also used in other section of the app)

I'd say flat approach is a bit flaky (but case to case basis).